server/utils/applications/utils.test.ts

```diff
@@ -603,10 +692,32 @@ describe('utils', () => {
     })
   })
 
-  describe('withdrawCell', () => {
+  describe('actionsCell', () => {
     it('returns a link to withdraw the application', () => {
-      expect(withdrawCell('id')).toEqual({
-        html: linkTo(paths.applications.withdraw.new, { id: 'id' }, { text: 'Withdraw' }),
+      const application = applicationFactory.build()
+      const actionItems: Array<string> = []
+      const withDrawLink = linkTo(paths.applications.withdraw.new, { id: application.id }, { text: 'Withdraw' })
+      actionItems.push(withDrawLink)
+      const actionLinks = `<ul class="govuk-list">${actionItems.map(actionItem => `<li>${actionItem}</li>`).join('')}</ul>`
+      expect(actionsCell(application)).toEqual({
+        html: actionLinks,
+      })
+    })
+
+    it('returns a link to withdraw and request for placement of the application', () => {
+      const application = applicationFactory.build({ status: 'awaitingPlacement' })
+      const actionItems: Array<string> = []
+      const withDrawLink = linkTo(paths.applications.withdraw.new, { id: application.id }, { text: 'Withdraw' })
+      actionItems.push(withDrawLink)
+      const requestForPlacementLink = linkTo(
+        placementApplicationPaths.placementApplications.create,
+        { id: application.id },
+        { text: 'Request for placement' },
+      )
+      actionItems.push(requestForPlacementLink)
+      const actionLinks = `<ul class="govuk-list">${actionItems.map(actionItem => `<li>${actionItem}</li>`).join('')}</ul>`
+      expect(actionsCell(application)).toEqual({
+        html: actionLinks,
```

Comment:
Sorry, 'granular' probably wasn't the right word in my earlier comment! The problem now is that this is essentially reimplementing the logic of the actual method in the test, and testing the logic/implementation rather than the outcome

For a utils method like this, it's probably best to use an explicit object literal in the expected value, rather than relying on the same logic to build the expected value and the output. Something like this for the two tests would avoid these issues and document the conditional expectations:

```ts
  describe('actionsCell', () => {
    it('returns a link to withdraw the application', () => {
      const application = applicationFactory.build({ id: 'an-application-id' })

      expect(actionsCell(application)).toEqual({
        html: '<ul class="govuk-list"><li><a href="/applications/an-application-id/withdrawals/new"  >Withdraw</a></li></ul>',
      })
    })

    it('returns a link to withdraw and request for placement of the application', () => {
      const application = applicationFactory.build({ id: 'an-application-id', status: 'awaitingPlacement' })

      expect(actionsCell(application)).toEqual({
        html: '<ul class="govuk-list"><li><a href="/applications/an-application-id/withdrawals/new"  >Withdraw</a></li><li><a href="/placement-applications"  >Request for placement</a></li></ul>',
      })
    })
  })
```

Reply:
yah we already have tests for linkTo thats why just added for actions